node/test/health.js

```diff
@@ -0,0 +1,170 @@
+    });
+
+    var client = cluster.channels[1].subChannels.server;
+
+    tchannelAsThrift.send(client.request({
+        serviceName: 'server',
+        hasNoParent: true
+    }), 'Chamber::echo', null, {
+        value: 10
+    }, function onResponse(err, res) {
+        assert.ifError(err);
+
+        assert.ok(res.ok);
+        assert.equal(res.headers.as, 'thrift');
+        assert.equal(res.body, 10);
+        assert.end();
+    });
+});
+
+allocCluster.test('health check works in good scenarios', {
+    numPeers: 2
+}, function t(cluster, assert) {
+    makeTChannelThriftServer(cluster, {
+        okResponse: true,
+        good: true
+    });
+
+    var client = cluster.channels[1].subChannels.server;
+    var source = fs.readFileSync(path.join(__dirname, '../as/meta.thrift'), 'utf8');
+    var healthThrift = new TChannelAsThrift({
+        source: source,
+        logParseFailures: false
+    });
+
+    healthThrift.send(client.request({
+        serviceName: 'server',
+        hasNoParent: true
+    }), 'Meta::health', null, {}, function onResponse(err, res) {
+        if (err) {
+            assert.end(false);
+        }
+
+        assert.ok(res && res.ok && res.body.ok, 'res should be ok');
+        assert.end();
+    });
+});
+
+allocCluster.test('health check works in bad scenarios', {
+    numPeers: 2
+}, function t(cluster, assert) {
+    makeTChannelThriftServer(cluster, {
+        okResponse: true,
+        good: false
+    });
+
+    var client = cluster.channels[1].subChannels.server;
+    var source = fs.readFileSync(path.join(__dirname, '../as/meta.thrift'), 'utf8');
+    var healthThrift = new TChannelAsThrift({
+        source: source,
+        logParseFailures: false
+    });
+
+    healthThrift.send(client.request({
+        serviceName: 'server',
+        hasNoParent: true
+    }), 'Meta::health', null, {}, function onResponse(err, res) {
+        if (err) {
+            assert.end(false);
+        }
+
+        assert.ok(res && res.ok && !res.body.ok, 'res body should not be ok');
+        assert.equals(res.body.message, 'bad thing happened!', 'health check message should be returned');
+        assert.end();
+    });
+});
+
+function healthGood() {
+    return {
+        ok: true
+    };
+}
+
+function healthBad() {
+    return {
+        ok: false,
+        message: 'bad thing happened!'
+    };
+}
+
+function makeTChannelThriftServer(cluster, opts) {
+    var server = cluster.channels[0].makeSubChannel({
+        serviceName: 'server'
+    });
+
+    cluster.channels[1].makeSubChannel({
+        serviceName: 'server',
+        peers: [
+            cluster.channels[0].hostPort
+        ],
+        requestDefaults: {
+            headers: {
+                cn: 'wat'
+            }
+        }
+    });
+
+    var options = {
+        isOptions: true
+    };
+
+    var health = opts.good ? healthGood : healthBad;
+
+    var tchannelAsThrift = new TChannelAsThrift({
+        source: opts.thriftText || globalThriftText,
```

Comment:
Please name this `thriftSource` or simply `source`. Don’t mix `text` and `source` for the same semantics.